### DISCUSSION TITLE
Adding more mouse functionality to pyqtgrapgh (QToolTip that displays the XY cords of mouse pointer in relation to graph item)

### DISCUSSION THREAD

**Discussion by mikelemo** [pyqtgraph/pyqtgraph#2220] [Q&A]
Hi,

I'd like to add some more functionality around pyqtgraph as such when you hover over the graph item the widget reads the x cords of the mouse in relation to the plot item and display X and Y coordinates according to where the pointer inform of some customized QToolTip Widget that'll follow the mouse.
Also after achieving that I'll be interested in making a point selector of sort with the left click mouse.

Now my question is how would one do that without overwriting or copying the exiting library files source code? 

I've tryed that making my own instance of class that inherits from pg.PlotWidget copied mouseMoveEvent() function source code from GraphicsView.py of pg library and managed to get printout of mouse cursor of *The application* every time the mouse overs over to plot widget and I'd like to get the position of the cursor the *The ViewBox* of the plot widget or in relation to it, and running pyqtgraph with Qt6.1 mapToScene(lpos) doesn't work and outputs the following error:

> Traceback (most recent call last):
  File "C:\Users\User\AppData\Roaming\Python\Python39\site-packages\pyqtgraph\widgets\GraphicsView.py", line 367, in mouseMoveEvent
    self.sigSceneMouseMoved.emit(self.mapToScene(lpos))
TypeError: arguments did not match any overloaded call:
  mapToScene(self, QPoint): argument 1 has unexpected type 'QPointF'
  mapToScene(self, QRect): argument 1 has unexpected type 'QPointF'
  mapToScene(self, QPolygon): argument 1 has unexpected type 'QPointF'
  mapToScene(self, QPainterPath): argument 1 has unexpected type 'QPointF'
  mapToScene(self, int, int): argument 1 has unexpected type 'QPointF'
  mapToScene(self, int, int, int, int): argument 1 has unexpected type 'QPointF'

**Comment by j9ac9k** [comment]
Hi @mikelemo 

The coordinate systems in the Qt's Graphics View can be quite confusing, and I hope I don't out myself when I admit that I don't have a full grasp on them either.  Changing the `sigSceneMouseMoved` to not emit a floating-point precision coordinate I'm not sure is in the cards.   That said, you can definitely make this work, as I have similar functionality in my own application:

In my case, you don't want the scene coordiantes... I have a method that does the following


```python
def invokeToolTip(self, event):
    if self.sceneBoundingRect().contains(event):
        if not self.isVisible():
            return None
        point = self.getViewWidget().mapToGlobal(event.toPoint())
        QToolTip.showText(point, "my text!")
```

A word of warning about `QToolTip` 's.  ...well two warnings

1. I've never been able to set the dismiss duration to something "long", no matter what I set it to, after a few seconds, the QToolTip goes away.
2. The QToolTip is like a mini windows/widget on top of your window, so not all screen-sharing apps will capture it.

    
Here is an example of how I use QToolTip's in an application I maintain.

https://user-images.githubusercontent.com/646398/157117507-ef7e62ac-4a56-4a23-829d-0f175ab7c6d6.mov

**Reply by mikelemo** [reply]
Hello @j9ac9k, 
If you said you never managed to toggle on the tool tip curosr I'll probably be looking for some other method of displaying the pointed cords with labels or some custom widget or something that'd less of a problem, The video you showed in the last example is in the direction of where I'm trying to go but in order to have a reliable application (for my case atleast) I need to be able to locate the location of the mouse cursor in relation to just the ViewBox rather than the whole application since the size of pixels is dynamic per axis width a screen width in general, 
I was just wondering is there's a low intensity way of receiving the cursor points in a ratio way or some method that'd allow me to relate the the ViewBox and the displayed data in it without having to read the axis, title and other widths as well as dynamically read the screen size in pixels to just find my XY cords of where the mouse sits in the graphbox\ViewBox.
Other than a single simple Qt app I'm not that experienced with Qt framework and or python, mostly embedded apps in C.
